### PR TITLE
Handle successful test with failed assertions during Junit report generation

### PR DESCRIPTION
**Pull request by acecilia**
👋 

Handle successful test with failed assertions during Junit report generation. See code for details.

Context: I experienced the edge case of a test with a failed assertion, but success status:

Screenshot 2024-10-31 at 03 55 36

### CODE REVIEW

File: Tests/XCTestHTMLReportTests/JUnitReportTests.swift [hunk 1]
```diff
@@ -76,8 +76,8 @@ final class JUnitReportTests: XCTestCase {
         let string = jUnitReport.xmlString
         let expectedString = #"""
         <?xml version='1.0' encoding='UTF-8'?>
-        <testsuites name='JUnitReportName&lt;&apos;&amp;\&gt;' tests='1' failures='1'>
-          <testsuite name='JUnitReportTestSuiteName&lt;&apos;&amp;\&gt;' tests='1' failures='1'>
+        <testsuites name='JUnitReportName&lt;&apos;&amp;\&gt;' tests='1' failures='1' skipped='0'>
+          <testsuite name='JUnitReportTestSuiteName&lt;&apos;&amp;\&gt;' tests='1' failures='1' skipped='0'>
```
Comment (acecilia) [inline Tests/XCTestHTMLReportTests/JUnitReportTests.swift:80]: These changes I believe are leftovers from previous PR [here](https://github.com/XCTestHTMLReport/XCTestHTMLReport/pull/362)

**Comment by acecilia** [comment]
@tylervick any chance you could have a look? 🙏

**Comment by acecilia** [comment]
Any chance to get this through @tylervick ?
